### PR TITLE
Update adventure.py

### PR DESCRIPTION
**Pull request by Ririsama**
passive bonus for elf human fairy dwarf and valkyrie

### CODE REVIEW

File: adventure/adventure.py [hunk 1]
```diff
@@ -4161,20 +4259,60 @@ def E(t: str) -> str:
...
+                        cp_bonus_list_names.append(self.E(user.display_name))
+                    else:
+                        xp_bonus += bonus
+                        xp_bonus_list_names.append(self.E(user.display_name))
+        if len(xp_bonus_list_names) > 0:
+            bonus = max(1, int(xp * min(100, xp_bonus) / 100))
+            xp += bonus        
+            phrase += (
+                f"\n{bold(humanize_list(xp_bonus_list_names))} led this battle with great panache... "
+                f"*[+{bonus * len(userlist)} xp for the party!]*"
```
Comment (oddslol) [inline adventure/adventure.py]: This is going to make it seem like a huge bonus... if you have like 10 users and 200 xp then it'll say "[+2000xp for the party]", which is technically true... but best just to say "[+200xp each]" ?

File: adventure/adventure.py [hunk 2]
```diff
@@ -4117,8 +4215,8 @@ def E(t: str) -> str:
             modif = 0.5
         xp = max(1, round(amount))
         cp = max(1, round(amount * modif))
+        xp_bonus, cp_bonus, phrase = self._reward_bonus(ctx, userlist, xp, cp)
```
Comment (oddslol) [inline adventure/adventure.py]:
```suggestion
        xp_bonus, cp_bonus, phrase = await self._reward_bonus(ctx, userlist, xp, cp)
```

File: adventure/adventure.py [hunk 3]
```diff
@@ -3182,15 +3229,20 @@ def E(t: str) -> str:
...
-                repair_list.append([users, " to repair their gear.\n", " to have their gear repaired...\n"])
-                options = [
-                    f"No amount of diplomacy or valiant fighting could save you.\n",
-                    f"This challenge was too much for one hero.\n",
-                    f"You tried your best, but the group couldn't succeed at their attempt.\n"
-                ]
-                text = random.choice(options)
+                if escape:
+                    text = escape_txt
+                    repair_list.append([remaining_users, " to repair their gear.\n", " to have their gear repaired...\n"])
```
[the remark this answers is not in the document]
Comment (Ririsama) [reply adventure/adventure.py]: escape can't totally fail and return full list in remaining_users as at least one pegasus will be called by a valkyrie

File: adventure/adventure.py [hunk 4]
```diff
@@ -3273,15 +3325,20 @@ def E(t: str) -> str:
...
-                repair_list.append([users, " to repair their gear.\n", " to have their gear repaired...\n"])
-                options = [
-                    f"No amount of diplomacy or valiant fighting could save you.\n",
-                    f"This challenge was too much for the group.\n",
-                    f"You tried your best, but couldn't succeed.\n"
-                ]
-                text = random.choice(options)
+                if escape:
+                    text = escape_txt
+                    repair_list.append([remaining_users, " to repair their gear.\n", " to have their gear repaired...\n"])
```
[the remark this answers is not in the document]
Comment (Ririsama) [reply adventure/adventure.py]: same remark

File: adventure/adventure.py [hunk 5]
```diff
@@ -3290,6 +3347,47 @@ def E(t: str) -> str:
...
+            escape = True
+            if len(escape_users) == 1:
+                attrib1 = "her"
+                attrib2 = ""
+            else:
+                attrib1 = "their"
+                attrib2 = "s"
+            escape_txt += f"{humanize_list(escape_users)} called {pegasus_nb} Pegasus with {attrib} magic horn{attrib2}, right in time to escape!\n"
+            shuffle_users = random.shuffle(users)
+            For user in shuffle_users:
```
[the remark this answers is not in the document]
Comment (Ririsama) [reply adventure/adventure.py]: corrected with your proposal

File: adventure/adventure.py [hunk 6]
```diff
@@ -3290,6 +3347,47 @@ def E(t: str) -> str:
...
+            try:
+                c = await Character._from_json(self.config, user)
+            except Exception:
+                log.error("Error with the new character sheet", exc_info=True)
+                continue
+            if c.race == "valkyrie":
+                roll = random.randint(1, 100)
+                chance = max(1, int(c.lvl * 25 / 50))
+                if roll in range(1, chance):
+                    escape_chance += c.lvl * 2
```
[the remark this answers is not in the document]
Comment (Ririsama) [reply adventure/adventure.py]: escape_chance isnt between 0 and 1, i need to divide it by 100 in pegasus_nb formula, you're right.

File: adventure/adventure.py [hunk 7]
```diff
@@ -3290,6 +3347,47 @@ def E(t: str) -> str:
...
+        if len(escape_users) > 0:
+            pegasus_nb = min(max(1, round(escape_chance * len(users)), len(users))
+            escape = True
+            if len(escape_users) == 1:
+                attrib1 = "her"
+                attrib2 = ""
+            else:
+                attrib1 = "their"
+                attrib2 = "s"
+            escape_txt += f"{humanize_list(escape_users)} called {pegasus_nb} Pegasus with {attrib} magic horn{attrib2}, right in time to escape!\n"
```
[the remark this answers is not in the document]
Comment (Ririsama) [reply adventure/adventure.py]: fixed

**Pull request merged**